widgetssdk/src/main/java/com/glia/widgets/engagement/EngagementRepositoryImpl.kt

```diff
@@ -220,13 +220,14 @@ internal class EngagementRepositoryImpl(
             unsubscribeFromEvents(it)
             resetState()
 
-            it.end { ex -> ex?.also { Logger.d(TAG, "Ending engagement failed") } }
+            //End the current engagement only when it is ended by the visitor.
+            if (endedBy == EndedBy.VISITOR) {
+                it.end { ex -> ex?.also { Logger.d(TAG, "Ending engagement failed") } }
+            }
 
-            val state = State.EngagementEnded(
-                it.isCallVisualizer,
-                endedBy,
-                it.state.actionOnEnd
-            ) { onSuccess, onError -> fetchSurvey(it, onError, onSuccess) }
+            val state = State.EngagementEnded(it.isCallVisualizer, endedBy, it.state.actionOnEnd) { onSuccess, onError ->
+                fetchSurvey(it, onError, onSuccess)
+            }
```

Comment:
So `state` will be set to `State.EngagementEnded` even in case `it.end` was not called (`endedBy` is not `EndedBy.VISITOR`). Looks unlogical to me. 🤔

Reply:
I think I was missing that in all other cases engagement is already ended. This function needs to call `it.end()` ONLY if visitor initiated this action(event). So should be fine.